state/indexer/block/kv/kv.go

```diff
@@ -86,6 +86,20 @@ func (idx *BlockerIndexer) Index(bh types.EventDataNewBlockEvents) error {
 	return batch.WriteSync()
 }
 
+func (idx *BlockerIndexer) Prune(retainHeight int64) {
+	for height := int64(0); height < retainHeight; height++ {
+		key, err := heightKey(height)
+		if err != nil {
+			panic(err)
+		}
+		err = idx.store.Delete(key)
```

Comment:
This looks massively inefficient. If this is performed regularly, the amount of work going through already pruned indices will grow as the retain height grows.

Can the store keys that are actually present be enumerated in an ordered way, from earliest height towards the latest?

Reply:
What can be done is something similar to what we do in the pruning mechanism - remember the last retain height and prune between the last retain height and the retain height. See how it's retrieved and used [here](https://github.com/cometbft/cometbft/blob/0dd35c933a1ccea1a2e439a8a74b5fd5cc9103fd/state/store.go#L395C22-L395C40).